tests/Unit/Evolution/Systems/Cce/AnalyticSolutions/Test_TeukolskyWave.cpp

```diff
@@ -0,0 +1,297 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cmath>
+#include <complex>
+#include <cstddef>
+#include <limits>
+
+#include "DataStructures/ComplexDataVector.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/SpinWeighted.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/Cce/AnalyticSolutions/TeukolskyWave.hpp"
+#include "Evolution/Systems/Cce/BoundaryDataTags.hpp"
+#include "Evolution/Systems/Cce/Tags.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "Helpers/Evolution/Systems/Cce/AnalyticSolutions/AnalyticDataHelpers.hpp"
+#include "NumericalAlgorithms/Spectral/SwshCollocation.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Cce::Solutions {
+
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.Cce.TeukolskyWave", "[Unit][Cce]") {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<double> radius_dist{5.0, 10.0};
+  UniformCustomDistribution<double> parameter_dist{0.5, 2.0};
+  const double extraction_radius = radius_dist(gen);
+  const size_t l_max = 16;
+  // use a low frequency so that the dimensionless parameter in the metric is ~1
+  const double amplitude = parameter_dist(gen) / 10.0;
```

Comment:
If I'm reading the code right, this line is the only place you use `parameter_dist`. So why not draw from `{0.05, 0.2}` and avoid dividing by 10?